src/test/java/org/gridsuite/study/server/NetworkModificationTest.java

```diff
@@ -589,6 +589,95 @@ public void testNetworkModificationEquipment() throws Exception {
         wireMockUtils.verifyNetworkModificationPostWithVariant(stubId, bodyJson, NETWORK_UUID_STRING, VARIANT_ID_2);
     }
 
+    @Test
+    public void testCreateBattery() throws Exception {
+        String userId = "userId";
+        StudyEntity studyEntity = insertDummyStudy(UUID.fromString(NETWORK_UUID_STRING), CASE_UUID, "UCTE");
+        UUID studyNameUserIdUuid = studyEntity.getId();
+        UUID rootNodeUuid = getRootNode(studyNameUserIdUuid).getId();
+        NetworkModificationNode modificationNode1 = createNetworkModificationNode(studyNameUserIdUuid, rootNodeUuid,
+                UUID.randomUUID(), VARIANT_ID, "node 1", userId);
+        UUID modificationNode1Uuid = modificationNode1.getId();
+        NetworkModificationNode modificationNode2 = createNetworkModificationNode(studyNameUserIdUuid,
+                modificationNode1Uuid, UUID.randomUUID(), VARIANT_ID_2, "node 2", userId);
+        UUID modificationNode2Uuid = modificationNode2.getId();
+
+        Map<String, Object> body = new HashMap<>();
+        body.put("type", ModificationType.BATTERY_CREATION);
+        body.put("batteryId", "batteryId1");
+        body.put("batteryName", "batteryName1");
+        body.put("minActivePower", "100.0");
+        body.put("maxActivePower", "200.0");
+        body.put("activePowerSetpoint", "10.0");
+        body.put("reactivePowerSetpoint", "20.0");
+        body.put("voltageLevelId", "idVL1");
+        body.put("busOrBusbarSectionId", "idBus1");
+        String bodyJsonCreate = mapper.writeValueAsString(body);
+
+        // create generator on root node (not allowed)
+        mockMvc.perform(post(URI_NETWORK_MODIF, studyNameUserIdUuid, rootNodeUuid)
+                        .content(bodyJsonCreate).contentType(MediaType.APPLICATION_JSON)
+                        .header(USER_ID_HEADER, userId))
+                .andExpect(status().isForbidden());
```

Comment:
It would be helpful to add an assertion message to the andExpect() method to provide more context in case of a failure.